src/selectors/profile.js

```diff
@@ -170,6 +172,49 @@ export const getContentfulSpeedIndexProgress: Selector<
 export const getProfilerConfiguration: Selector<?ProfilerConfiguration> = state =>
   getMeta(state).configuration;
 
+/**
+ * TODO - These will eventually be stored in the profile, but for now
+ * define them here.
+ */
+export const getMarkerSchema: Selector<MarkerSchema[]> = () => [
+  {
+    name: 'GCSlice',
+    label: 'GCSlice',
+    display: ['marker-chart', 'marker-table', 'timeline-memory'],
+    data: [],
+  },
+  {
+    name: 'GCMajor',
+    label: 'GCMajor',
+    display: ['marker-chart', 'marker-table', 'timeline-memory'],
+    data: [],
+  },
+  {
+    name: 'GCMinor',
+    label: 'GCMinor',
+    display: ['marker-chart', 'marker-table', 'timeline-memory'],
+    data: [],
+  },
+  {
+    name: 'GCSlice',
+    label: 'GCSlice',
+    display: ['marker-chart', 'marker-table', 'timeline-memory'],
+    data: [],
+  },
+  {
+    name: 'CC',
+    label: 'Cycle Collect',
+    display: ['marker-chart', 'marker-table', 'timeline-memory'],
+    data: [],
+  },
+];
+
+export const getTimelineMemoryMarkerTypes: Selector<
+  Set<string>
+> = createSelector(getMarkerSchema, markerSchema =>
+  MarkerData.getMarkerTypesForDisplay(markerSchema, 'timeline-memory')
+);
```

Comment:
For future work: we'll possibly want to create all the sets for all displays in one go, instead of iterating the markerSchema several times, once for each display. This isn't a big deal now, but in busy profiles with a lot of different markers this may start to be a problem.
So we could have a selector `getMarkerTypesPerDisplay: Selector<Map<MarkerDisplayLocations, Set<string>>>`, or a keyed object instead of the Map, I don't really mind. Then we could have individual selectors for each time if you want too, for simplicity (dealing with Flow's checking of null value etc).

Reply:
Yeah, I'm looking more into filtering now, and each timeline has its own iteration check. This is existing behavior as far as I can tell. The schema adds a bit more complexity to these checks, especially in the form of additional function calls.